pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
+        varExons[exonName] = {"exonStart": exonStart,
+                              "exonEnd": exonEnd}
+        exonCount += 1
+
+    return varExons
+
+
+def getRefSpliceDonorBoundaries(variant):
+    '''
+    Given a variant, returns the splice donor boundaries 
+    (splice donor region is last 3 bases in exon and frist 6 bases in intron) 
+    for the variant's transcript in a dictionary with the format:
+    key = exon number, value = dictionary with donor start and donor end for exon
+    '''
+    varExons = getExonBoundaries(variant)
+    if variant["Gene_Symbol"] == "BRCA1":
+        del varExons["exon24"]
+    elif variant["Gene_Symbol"] == "BRCA2":
+        del varExons["exon27"]
+    varStrand = getVarStrand(variant)
+    donorBoundaries = {}
+    for exon in varExons.keys():
+        exonEnd = int(varExons[exon]["exonEnd"])
+        if varStrand == "+":
+            donorStart = exonEnd - 3 + 1
+            donorEnd = exonEnd + 6
+        else:
+            donorStart = exonEnd + 3
+            donorEnd = exonEnd - 6 + 1
+        donorBoundaries[exon] = {"donorStart": donorStart,
+                                 "donorEnd": donorEnd}
+
+    return donorBoundaries
 
 
-def getVarDict(variant):
+def getRefSpliceAcceptorBoundaries(variant):
+    '''
+    Given a variant, returns the splice acceptor boundaries
+    (splice acceptor region is 20 bases before exon and first 3 bases in exon)
+    for the variant's transcript in a dictionary with the format:
+    key = exon number, value = a dictionary with acceptor start and acceptor end for exon
+    '''
+    varExons = getExonBoundaries(variant)
+    if variant["Gene_Symbol"] == "BRCA1" or variant["Gene_Symbol"] == "BRCA2":
+        del varExons["exon1"]
+    varStrand = getVarStrand(variant)
+    acceptorBoundaries = {}
+    for exon in varExons.keys():
+        exonStart = int(varExons[exon]["exonStart"])
+        if varStrand == "+":
+            acceptorStart = exonStart - 20 + 1
+            acceptorEnd = exonStart + 3
+        else:            
+            acceptorStart = exonStart + 20
+            acceptorEnd = exonStart - 3 + 1
+        acceptorBoundaries[exon] = {"acceptorStart": acceptorStart,
+                                    "acceptorEnd": acceptorEnd}
+
+    return acceptorBoundaries
+
+
+def varInExon(variant):
+    '''
+    Given a variant, determines if variant genomic position is inside transcript boundaries
+    AND if variant is in an exon
+    Returns true if variant is in an exon
+    '''
+    varGenPos = int(variant["Pos"])
+    varExons = getExonBoundaries(variant)
+    varStrand = getVarStrand(variant)
+    varOutBounds = varOutsideBoundaries(variant)
+    if varOutBounds == False:
+        for exon in varExons.keys():
+            exonStart = varExons[exon]["exonStart"]
+            exonEnd = varExons[exon]["exonEnd"]
+            if varStrand == "+":
+                if varGenPos > exonStart and varGenPos <= exonEnd:
+                    return True
```

Comment:
careful here... any time you hit a return statement inside of a for loop it will exit the for loop (otherwise known as shortcutting the loop). Is this intentional? I noticed there are many of these in this PR that will need to be looked over carefully.

also, it seems you'd only want to shortcut the for loop with a `return False` -- i.e. if any key falls outside the exon return false, otherwise all keys in the for loop pass and you can return true.

Reply:
I did intentionally mean to shortcut the loop. I thought it would be better for it to exit the loop as soon as it found that the genomic position was in an exon and then stop checking the rest of the exons. If it finished going through the loop then I wanted it to return false because then it meant that the genomic position was not in an exon. But I'm happy to change it if it makes more sense to do the opposite